### PR TITLE
fix: Simplify Contract Subscriptions block range

### PR DESCRIPTION
**Pull request by arcoraven**
## PR-Codex overview
This PR focuses on improving the handling of transaction receipts and event logs within a blockchain indexing worker. It includes enhancements to logging, error handling, and the estimation of block times, while also refactoring some code for better clarity and efficiency.

### Detailed summary
- Added logging for inserted receipts in `processTransactionReceiptsWorker.ts`.
- Changed `parseInt` to `Number.parseInt` in `manageChainIndexers.ts`.
- Simplified access to properties in `getChainIndexer.ts`.
- Implemented tests for `getBlockTimeSeconds` in `getBlockTime.test.ts`.
- Removed `CONTRACT_SUBSCRIPTIONS_DELAY_SECONDS` from environment variables in `env.ts`.
- Removed Redis dependency checks in `chainIndexerListener.ts`.
- Updated block time estimation logic in `chainIndexer.ts`.
- Refactored block timestamp retrieval in `getBlockTime.ts`.
- Improved error handling and logging throughout the worker tasks.
- Consolidated contract subscription handling in `chainIndexer.ts`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/utils/env.ts [hunk 1]
```diff
@@ -69,10 +69,6 @@ export const env = createEnv({
     SDK_BATCH_TIME_LIMIT: z.coerce.number().default(0),
     SDK_BATCH_SIZE_LIMIT: z.coerce.number().default(100),
     ENABLE_KEYPAIR_AUTH: boolEnvSchema(false),
-    CONTRACT_SUBSCRIPTIONS_DELAY_SECONDS: z.coerce
```
Comment (arcoraven) [inline src/utils/env.ts:72]: Experimental + undocumented. Not useful.

File: src/utils/indexer/getBlockTime.ts [hunk 2]
```diff
@@ -1,49 +1,26 @@
...
+  const latestBlock = await eth_getBlockByNumber(rpcRequest, {
+    blockTag: "latest",
+    includeTransactions: false,
+  });
+  const referenceBlock = await eth_getBlockByNumber(rpcRequest, {
+    blockNumber: latestBlock.number - BigInt(blocksToEstimate),
+    includeTransactions: false,
+  });
 
-export const getBlockTimeSeconds = async (chainId: number) => {
```
Comment (arcoraven) [inline src/utils/indexer/getBlockTime.ts:12]: The previous logic would add up all timestamps between the last 100 blocks. This should be the same as just taking the diff the current block and the n-100 block.

File: src/utils/indexer/getBlockTime.ts [hunk 3]
```diff
@@ -1,49 +1,26 @@
...
-import { logger } from "../logger";
+import { eth_getBlockByNumber, getRpcClient } from "thirdweb";
+import { getChain } from "../chain";
+import { thirdwebClient } from "../sdk";
 
-const KNOWN_BLOCKTIME_SECONDS = {
-  1: 12,
-  137: 2,
-} as Record<number, number>;
+export const getBlockTimeSeconds = async (
```
Comment (arcoraven) [inline src/utils/indexer/getBlockTime.ts:5]: Simplifies logic + moves to v5.

Previously this code was summing the timestamps for 100 blocks. Now it just takes the diff between the n and n-100 block, which should be equivalent.

File: src/worker/indexers/chainIndexerRegistry.ts [hunk 4]
```diff
@@ -1,67 +1,57 @@
...
-
-  const blocksIn5Seconds = Math.round((1 / blockTimeSeconds) * 5);
-  const maxBlocksToIndex = Math.max(
-    config.maxBlocksToIndex,
-    blocksIn5Seconds * 4,
-  );
-
-  // Compute block offset based on delay.
-  // Example: 10s delay with a 3s block time = 4 blocks offset
-  const toBlockOffset = env.CONTRACT_SUBSCRIPTIONS_DELAY_SECONDS
```
Comment (arcoraven) [inline src/worker/indexers/chainIndexerRegistry.ts:34]: Removing this. Unused.

File: src/worker/indexers/chainIndexerRegistry.ts [hunk 5]
```diff
@@ -1,67 +1,57 @@
...
-  const config = await getConfig();
-
-  // Estimate block time.
-  const blockTimeSeconds = await getBlockTimeSeconds(chainId);
-
-  const blocksIn5Seconds = Math.round((1 / blockTimeSeconds) * 5);
-  const maxBlocksToIndex = Math.max(
-    config.maxBlocksToIndex,
-    blocksIn5Seconds * 4,
-  );
```
Comment (arcoraven) [inline src/worker/indexers/chainIndexerRegistry.ts:30]: This was more complicated than it's worth. The goal was to find a good range of blocks to query. Honestly if we're realtime we're usually querying 5 seconds worth of blocks (2-20 depending on the block time). Hard coding this to a value greater than this but less than most eth_getLogs limits should suffice.

File: src/worker/tasks/chainIndexer.ts [hunk 6]
```diff
@@ -1,135 +1,122 @@
...
-  maxBlocksToIndex: number;
-  toBlockOffset: number;
-}) => {
-  const { chainId, maxBlocksToIndex, toBlockOffset } = args;
+// A reasonable block range that is within RPC limits.
+// The minimum job time is 1 second, so this value should higher than the # blocks per second
+// on any chain to allow catching up if delayed.
+const MAX_BLOCK_RANGE = 500;
 
-  const chainIndexerTask = async () => {
```
Comment (arcoraven) [inline src/worker/tasks/chainIndexer.ts:19]: Changes:
- This function no longer returns a function, it just executes the logic.
- Remove try/catch, already caught above.
- Move to v5 SDK

**Review by d4mr — APPROVED** [review]

**Pull request merged**